server/src/main/java/org/opensearch/Version.java

```diff
@@ -82,18 +85,25 @@ public static Version fromId(int id) {
         if (known != null) {
             return known;
         }
-        return fromIdSlow(id);
+        return fromIdSlow(id, false);
     }
 
-    private static Version fromIdSlow(int id) {
+    public static Version fromLegacyId(int id) {
```

Comment:
I think we should call this `fromLegacyESId` to match `LegacyESVersion` and to be specific. One day there might be more legacy.